src/hotspot/share/gc/g1/g1ConcurrentMark.cpp

```diff
@@ -2274,7 +2280,7 @@ bool G1CMTask::regular_clock_call() {
 
   // (5) We check whether we've reached our time quota. If we have,
   // then we abort.
-  double elapsed_time_ms = curr_time_ms - _start_time_ms;
+  double elapsed_time_ms = (double)(curr_time_ns - _start_cpu_time_ns) / NANOSECS_PER_MILLISEC;
```

Comment:
Maybe there should be a helper for this calculation, since there are several occurrences of it?